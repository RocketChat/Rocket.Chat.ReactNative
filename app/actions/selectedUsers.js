import * as types from './actionsTypes';

export function addUser(user) {
	return {
		type: types.SELECTED_USERS.ADD_USER,
		user
	};
}

export function removeUser(user) {
	return {
		type: types.SELECTED_USERS.REMOVE_USER,
		user
	};
}

export function reset() {
	return {
		type: types.SELECTED_USERS.RESET
	};
}

export function setLoading(loading) {
	return {
		type: types.SELECTED_USERS.SET_LOADING,
		loading
	};
}

