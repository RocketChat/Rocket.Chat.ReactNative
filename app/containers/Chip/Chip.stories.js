import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View, StyleSheet } from 'react-native';
import { Provider } from 'react-redux';

import { store } from '../../../storybook/stories';
import { ThemeContext } from '../../theme';
import { colors } from '../../lib/constants';
import Chip from './index';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: 'flex-start',
		padding: 16
	}
});

const theme = 'light';

const stories = storiesOf('Chip', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.addDecorator(story => <ThemeContext.Provider value={{ theme, colors: colors[theme] }}>{story()}</ThemeContext.Provider>);

stories.add('Short and Long Chip', () => (
	<>
		<View style={styles.container}>
			<Chip avatar='rocket.cat' text={'Rocket.Cat'} />
			<Chip avatar='rocket.cat' text={'Short'} />
			<Chip text='Without Avatar' />
			<Chip avatar='rocket.cat' text='Without Icon' />
			<Chip text='Without Avatar and Icon' />
		</View>
	</>
));
