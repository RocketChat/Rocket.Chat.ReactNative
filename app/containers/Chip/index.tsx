import React from 'react';
import { Pressable, StyleSheet, View, Text, StyleProp, ViewStyle } from 'react-native';

import { useTheme } from '../../theme';
import { CustomIcon } from '../CustomIcon';
import sharedStyles from '../../views/Styles';
import Avatar from '../Avatar';

const styles = StyleSheet.create({
	pressable: {
		paddingHorizontal: 8,
		marginRight: 8,
		borderRadius: 4,
		justifyContent: 'center',
		maxWidth: 192
	},
	container: {
		flexDirection: 'row',
		alignItems: 'center'
	},
	avatar: {
		marginRight: 8,
		marginVertical: 8
	},
	textContainer: {
		marginRight: 8,
		maxWidth: 120
	},
	name: {
		fontSize: 16,
		...sharedStyles.textMedium
	}
});

export interface IChip {
	avatar?: string;
	text: string;
	onPress?: Function;
	testID?: string;
	style?: StyleProp<ViewStyle>;
}

const Chip = ({ avatar, text, onPress, testID, style }: IChip) => {
	const { colors } = useTheme();

	return (
		<Pressable
			testID={testID}
			style={({ pressed }) => [
				styles.pressable,
				{
					backgroundColor: pressed ? colors.bannerBackground : colors.auxiliaryBackground
				},
				style
			]}
			disabled={!onPress}
			onPress={() => onPress?.()}
			android_ripple={{
				color: colors.bannerBackground
			}}
		>
			<View style={styles.container}>
				{avatar ? <Avatar text={avatar} size={28} style={styles.avatar} /> : null}
				<View style={styles.textContainer}>
					<Text style={[styles.name, { color: colors.bodyText }]} numberOfLines={1}>
						{text}
					</Text>
				</View>
				{onPress ? <CustomIcon name='close' size={16} color={colors.auxiliaryTintColor} /> : null}
			</View>
		</Pressable>
	);
};

export default Chip;
