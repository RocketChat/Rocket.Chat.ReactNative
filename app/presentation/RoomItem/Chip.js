import React from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import { themes } from '../../constants/colors';

const Chip = React.memo(({ name, theme }) => (
	<View style={{ backgroundColor: themes[theme].auxiliaryText, padding: 8 }}>
		<Text
			style={[
				styles.title,
				{ color: themes[theme].bodyText }
			]}
			numberOfLines={1}
		>
			{name}
		</Text>
	</View>
));

Chip.propTypes = {
	name: PropTypes.string,
	theme: PropTypes.string
};

export default Chip;
