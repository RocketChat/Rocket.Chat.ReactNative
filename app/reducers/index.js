import { combineReducers } from 'redux';
import settings from './reducers';
import login from './login';
import meteor from './connect';
import messages from './messages';
import room from './room';
import rooms from './rooms';
import server from './server';
import selectedUsers from './selectedUsers';
import createChannel from './createChannel';
import app from './app';
import customEmojis from './customEmojis';
import activeUsers from './activeUsers';
import roles from './roles';
import starredMessages from './starredMessages';
import pinnedMessages from './pinnedMessages';
import mentionedMessages from './mentionedMessages';
import snippetedMessages from './snippetedMessages';
import roomFiles from './roomFiles';
import sortPreferences from './sortPreferences';

export default combineReducers({
	settings,
	login,
	meteor,
	messages,
	server,
	selectedUsers,
	createChannel,
	app,
	room,
	rooms,
	customEmojis,
	activeUsers,
	roles,
	starredMessages,
	pinnedMessages,
	mentionedMessages,
	snippetedMessages,
	roomFiles,
	sortPreferences
});
