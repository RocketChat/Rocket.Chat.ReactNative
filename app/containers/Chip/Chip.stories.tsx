import React from 'react';
import { StyleSheet, View } from 'react-native';

import Chip, { IChip } from './index';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: 'flex-start',
		padding: 16
	}
});

export default {
	title: 'Chip'
};

const ChipWrapped = ({ avatar, text, onPress, testID, style }: IChip) => (
	<View style={styles.container}>
		<Chip avatar={avatar} text={text} onPress={onPress} testID={testID} style={style} />
	</View>
);

export const ChipText = () => <ChipWrapped avatar='rocket.cat' text={'Rocket.Cat'} onPress={() => {}} />;

export const ChipWithShortText = () => <ChipWrapped avatar='rocket.cat' text={'Short'} onPress={() => {}} />;

export const ChipWithoutAvatar = () => <ChipWrapped text={'Without Avatar'} onPress={() => {}} />;

export const ChipWithoutIcon = () => <ChipWrapped avatar='rocket.cat' text='Without Icon' />;

export const ChipWithoutAvatarAndIcon = () => <ChipWrapped text='Without Avatar and Icon' />;
