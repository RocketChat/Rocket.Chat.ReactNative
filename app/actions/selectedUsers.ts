import { Action } from 'redux';

import { ISelectedUser } from '../reducers/selectedUsers';
import * as types from './actionsTypes';

type TUser = {
	user: ISelectedUser;
};

type TAction = Action & TUser;

interface ISetLoading extends Action {
	loading: boolean;
}

export type TActionSelectedUsers = TAction & ISetLoading;

export function addUser(user: ISelectedUser): TAction {
	return {
		type: types.SELECTED_USERS.ADD_USER,
		user
	};
}

export function removeUser(user: ISelectedUser): TAction {
	return {
		type: types.SELECTED_USERS.REMOVE_USER,
		user
	};
}

export function reset(): Action {
	return {
		type: types.SELECTED_USERS.RESET
	};
}

export function setLoading(loading: boolean): ISetLoading {
	return {
		type: types.SELECTED_USERS.SET_LOADING,
		loading
	};
}
