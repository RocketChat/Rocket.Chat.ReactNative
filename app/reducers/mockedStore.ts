import { applyMiddleware, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';

import reducers from '.';

const enhancers = compose(applyMiddleware(createSagaMiddleware()));
export const mockedStore = createStore(reducers, enhancers);
