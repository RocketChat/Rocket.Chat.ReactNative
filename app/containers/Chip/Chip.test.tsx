import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import Chip, { IChip } from '.';
import { ISelectedUser } from '../../reducers/selectedUsers';
import { mockedStore as store } from '../../reducers/mockedStore';

const onPressMock = jest.fn((item: any) => item);

const testChip = {
	testID: 'test-chip-id',
	item: { fname: 'rocket.chat', name: 'rocket.chat' } as ISelectedUser,
	onPress: onPressMock
};

const Render = ({ testID, text, avatar, onPress }: IChip) => (
	<Provider store={store}>
		<Chip testID={testID} text={text} onPress={onPress} avatar={avatar} />
	</Provider>
);

describe('Chips', () => {
	it('should render the Chip component', () => {
		const { findByTestId } = render(
			<Render
				text={testChip.item.fname}
				avatar={testChip.item.name}
				testID={testChip.testID}
				onPress={() => testChip.onPress(testChip.item)}
			/>
		);

		expect(findByTestId(testChip.testID)).toBeTruthy();
	});
	it("should not call onPress if it's not passed", async () => {
		const { findByTestId } = render(<Render text={testChip.item.fname} avatar={testChip.item.name} testID={testChip.testID} />);

		const component = await findByTestId(testChip.testID);
		fireEvent.press(component);
		expect(onPressMock).not.toHaveBeenCalled();
	});
	it('should tap Chip and return item', async () => {
		const { findByTestId } = render(
			<Render
				text={testChip.item.fname}
				avatar={testChip.item.name}
				testID={testChip.testID}
				onPress={() => testChip.onPress(testChip.item)}
			/>
		);

		const component = await findByTestId(testChip.testID);
		fireEvent.press(component);
		expect(onPressMock).toHaveBeenCalled();
		expect(onPressMock).toHaveReturnedWith(testChip.item);
	});
});
