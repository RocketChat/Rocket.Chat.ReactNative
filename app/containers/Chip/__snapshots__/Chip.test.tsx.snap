// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story snapshots: Chip ChipText snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 4,
          "justifyContent": "center",
          "marginRight": 8,
          "maxWidth": 192,
          "paddingHorizontal": 8,
        },
        {
          "backgroundColor": "#F2F3F5",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 8,
            "maxWidth": 120,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "left",
              },
              {
                "color": "#2F343D",
              },
            ]
          }
        >
          Rocket.Cat
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F343D",
              "fontSize": 32,
            },
            [
              {
                "lineHeight": 32,
              },
              undefined,
            ],
            {
              "fontFamily": "custom",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Story snapshots: Chip ChipWithShortText snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 4,
          "justifyContent": "center",
          "marginRight": 8,
          "maxWidth": 192,
          "paddingHorizontal": 8,
        },
        {
          "backgroundColor": "#F2F3F5",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 8,
            "maxWidth": 120,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "left",
              },
              {
                "color": "#2F343D",
              },
            ]
          }
        >
          Short
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F343D",
              "fontSize": 32,
            },
            [
              {
                "lineHeight": 32,
              },
              undefined,
            ],
            {
              "fontFamily": "custom",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Story snapshots: Chip ChipWithoutAvatar snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 4,
          "justifyContent": "center",
          "marginRight": 8,
          "maxWidth": 192,
          "paddingHorizontal": 8,
        },
        {
          "backgroundColor": "#F2F3F5",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 8,
            "maxWidth": 120,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "left",
              },
              {
                "color": "#2F343D",
              },
            ]
          }
        >
          Without Avatar
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F343D",
              "fontSize": 32,
            },
            [
              {
                "lineHeight": 32,
              },
              undefined,
            ],
            {
              "fontFamily": "custom",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Story snapshots: Chip ChipWithoutAvatarAndIcon snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 4,
          "justifyContent": "center",
          "marginRight": 8,
          "maxWidth": 192,
          "paddingHorizontal": 8,
        },
        {
          "backgroundColor": "#F2F3F5",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 8,
            "maxWidth": 120,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "left",
              },
              {
                "color": "#2F343D",
              },
            ]
          }
        >
          Without Avatar and Icon
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Story snapshots: Chip ChipWithoutIcon snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "borderRadius": 4,
          "justifyContent": "center",
          "marginRight": 8,
          "maxWidth": 192,
          "paddingHorizontal": 8,
        },
        {
          "backgroundColor": "#F2F3F5",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 8,
            "maxWidth": 120,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "left",
              },
              {
                "color": "#2F343D",
              },
            ]
          }
        >
          Without Icon
        </Text>
      </View>
    </View>
  </View>
</View>
`;
